src/Conversations/ConversationLoader.php

```diff
@@ -43,7 +43,11 @@ public function load()
         }
 
         if ($this->shouldLoadResource(ConversationLinks::THREADS)) {
-            $threads = $this->loadResources(Thread::class, ConversationLinks::THREADS);
+            $threadFactory = new ThreadFactory();
+            $threads = $this->loadResources(function (array $data) use ($threadFactory) {
+                return $threadFactory->make($data['type'], $data);
+            }, ConversationLinks::THREADS);
```

Comment:
@tompedals Thanks for making the deserializer accept a closure, that was brilliant :)